cmd/cluster/aws/create.go

```diff
@@ -55,6 +54,21 @@ func NewCreateCommand(opts *core.CreateOptions) *cobra.Command {
 			defer cancel()
 		}
 
+		if util.IsEmpty(opts.CredentialSecretName) {
+			if err := util.IsRequiredOption("aws-creds", opts.AWSPlatform.AWSCredentialsFile); err != nil {
+				return err
+			}
+			if err := util.IsRequiredOption("pull-secret", opts.PullSecretFile); err != nil {
+				return err
+			}
+		} else {
+			//Check the secret exists now, otherwise stop
+			opts.Log.Info("Retreiving credentials secret", "namespace", opts.Namespace, "name", opts.CredentialSecretName)
```

Comment:
This check is done now instead of when we use the secret data, as it allows for a clear error message when the secret is missing. Otherwise the error is wrapped in a different error message and is much harder to read.